src/org/jitsi/impl/neomedia/transform/RetransmissionRequesterImpl.java

```diff
@@ -201,369 +79,44 @@ else if (Constants.RTX.equalsIgnoreCase(format.getEncoding()))
-                if (nextRequestAt == -1 || nextRequestAt - now > 0)
-                {
-                    try
-                    {
-                        if (nextRequestAt == -1)
-                        {
-                            thread.wait(1000);
-                            continue;
-                        }
-                        else
-                        {
-                            thread.wait(nextRequestAt - now);
-                        }
-                    }
-                    catch (InterruptedException ie)
-                    {
-                        break;
-                    }
-                }
-                //else the time has already come
-            }
-
-            if (!enabled || senderSsrc == -1)
-                continue;
-
-            synchronized (requesters)
-            {
-                for (Map.Entry<Long, Requester> entry : requesters.entrySet())
-                {
-                    Requester requester = entry.getValue();
-                    Set<Integer> missingPackets = requester.getMissing();
-                    if (missingPackets != null && !missingPackets.isEmpty())
-                    {
-                        packetsToRequest.put(requester.ssrc, missingPackets);
-                    }
-                }
-            }
-
-            for (Map.Entry<Long, Set<Integer>> entry : packetsToRequest.entrySet())
-            {
-                long sourceSsrc = entry.getKey();
-                NACKPacket nack
-                    = new NACKPacket(senderSsrc, sourceSsrc, entry.getValue());
-                RawPacket pkt;
-
-                try
-                {
-                    pkt = nack.toRawPacket();
-                }
-                catch (IOException ioe)
-                {
-                    pkt = null;
-                    logger.warn("Failed to create a NACK packet: " + ioe);
-                }
-
-                if (pkt != null)
-                {
-                    try
-                    {
-                        if (logger.isTraceEnabled())
-                        {
-                            logger.trace("Sending a NACK: " + nack);
-                        }
-                        stream.injectPacket(
-                                pkt,
-                                /* data */ false,
-                                /* after */ null);
-                    }
-                    catch (TransmissionFailedException e)
-                    {
-                        logger.warn(
-                                "Failed to inject packet in MediaStream: ", e.getCause());
-                    }
-                }
-            }
-
-            packetsToRequest.clear();
-        }
-
-    }
-
-
-    /**
-     * Implements {@link TransformEngine#getRTPTransformer()}.
-     */
-    @Override
     public PacketTransformer getRTPTransformer()
     {
-        return this;
+        return this.retransmissionRequesterDelegate.getRTPTransformer();
```

Comment:
I think this hooks up the delagate as the transformer, bypassing RetransmissionRequesterImpl#reverseTransform()

Reply:
I think it would be best if all TransformEngine and PacketTransformer logic is handled in RetransmissionRequesterImpl, while the delegate can only provide the minimal needed API (e.g. packetReceived(RawPacket))

Reply:
> I think this hooks up the delagate as the transformer, bypassing 
> RetransmissionRequesterImpl#reverseTransform()

good catch, fixed

>I think it would be best if all TransformEngine and PacketTransformer logic is handled in > RetransmissionRequesterImpl, while the delegate can only provide the minimal needed API > (e.g. packetReceived(RawPacket))

think i see what you mean, i'll take a shot at this and see how it looks.

Reply:
i think that tweak looks pretty good